templates/search/blocks/search_results_timeline_century.html

```diff
@@ -1,129 +1,110 @@
-
-              // Get the canvas element for the chart
-              const ctx = document.getElementById("decade-counts-chart").getContext("2d");
-
-              // Create a new bar chart with click event handling
-              const chart = new Chart(ctx, {
-                type: "bar",
-                data: {
-                  labels: centuryLabels,
-                  datasets: [
-                    {
-                      label: "",
-                      data: chartData,
-                      backgroundColor: "rgba(219, 98, 91, 1)",
-                      borderColor: "rgba(255, 99, 132, 1)",
-                      borderWidth: 1,
-                    },
-                  ],
-                },
-                options: {
-                  onHover: (event, chartElement) =>{
-                    event.native.target.style.cursor = chartElement [0] ? 'pointer' : 'default';
-                  },
-                  // Adjust chart options as needed (e.g., scales, title)
-                  plugins: {
-                    legend: {
-                      display: false,  // This line hides the legend
-                    },
-                  },
-                  scales: {
-                    y:{ grid: {
-                        drawOnChartArea: false,
-                        color: "rgba(217 217, 214, 1)"
-                      },
-                      ticks: {
-                        display:false
-                      },
-                    },
-                    x: {
-                      position: 'top',
-                      grid: {
-                        drawBorder: false
-                      }
-                    },
-                  }
-                },
-              });
-
-              // Add click event listener to the chart canvas
-              chart.canvas.onclick = function(evt) {
-                const activePoints = chart.getElementsAtEventForMode(evt, 'index', false);
-                if (activePoints.length > 0) {
-                  // Get the clicked bar index
-                  const clickedIndex = activePoints[0].index;
-
-                  // Access the corresponding record count from chart data
-                  const recordCount = chartData[clickedIndex];
-                  
-                  window.location = `./?vis_view=timeline&timeline_type=decade&creation_date_from=${centuryLabels[clickedIndex]}#decade-counts-chart`;
-                }
-              };
-            })
-            .catch(error => {
-              console.error('Error fetching data:', error);
-              // Handle data fetching error (e.g., display an error message)
-            });
-
-  </script>
+  fetch('http://a08001c6c1d2e4c58bdf02589da33fca-0f850b505f873e8e.elb.eu-west-2.amazonaws.com/rosetta/data/search?aggs=decade,year,century&filter=group:community&filter=century:500&field=century:1900') // Replace with your server-side API endpoint
```

Comment:
Is it necessary to call the API exclusively? Also similarly called in other places.

Reply:
`&filter=century:500&field=century:1900`
One is filter, other is field - is this right?

Reply:
Could the JS be handled in a file of it own - for better readability and maintenance?

Reply:
The API domain is exposed in source. Should it be?

Reply:
Does the search term query, date filters, archive collection filters when applied also get applied to this API url?

Reply:
> Is it necessary to call the API exclusively? Also similarly called in other places.

The API gets called on page load for the three views.

Reply:
> `&filter=century:500&field=century:1900` One is filter, other is field - is this right? Why are values constrained to 500, 1900?

I need to  check this with Lee (K-Int), we may not need these parameters at the moment

Reply:
> The API domain is exposed in source. Should it be?

By compiling the js via webpack it will de-exposed the api call.(see above comment).

Reply:
> Does the search term query, date filters, archive collection filters when applied also get applied to this API url?

Not as yet but if we need to append these parameters to the url string, we can do via the js.

Reply:
> > The API domain is exposed in source. Should it be?
> 
> By compiling the js via webpack it will de-exposed the api call.(see above comment).

The API domain is exposed at GitHub source.

Reply:
> > Is it necessary to call the API exclusively? Also similarly called in other places.
> 
> The API gets called on page load for the three views.

What I meant was - why is the API called separately - can it not form a part of the same query for Timeline view?